File: dotnet/Allmark/RenderConsole/ConsoleCommentRenderer.cs
namespace Allmark.Render;

using Allmark.Types;

public static class ConsoleCommentRenderer
{
    public static OutputRenderer Create()
    {
        return new OutputRenderer
        {
            Name = "comment",
            Render = (node, state, first, last, decode) => Render(node, state),
        };
    }

    public static void Render(MarkdownNode node, RendererState state)
    {
        var style = Ansi.Blue;
        state.Output.Append($"{style}>>");
        RenderChildren.Execute(node, state);
        state.Output.Append($">>{Ansi.Reset}");
    }
}

File: zig/src/render-console/consoleCommentRenderer.zig
const std = @import("std");

const MarkdownNode = @import("../types/MarkdownNode.zig").MarkdownNode;
const ConsoleRendererState = @import("../types/RendererState.zig").RendererState;
const Renderer = @import("../types/Renderer.zig").Renderer;

pub const consoleCommentRenderer = Renderer{
    .name = "comment",
    .render = render,
};

pub fn render(node: *const MarkdownNode, state: *ConsoleRendererState, decode: ?bool) void {
    _ = decode;

    state.output.appendSlice(state.allocator, "<!--") catch unreachable;
    state.output.appendSlice(state.allocator, node.content) catch unreachable;
    state.output.appendSlice(state.allocator, "-->") catch unreachable;
}

pub const commentRenderer = struct {
    name: []const u8 = "comment",
    render: consoleCommentRenderer,
};
